### PR TITLE
Migrate from Travis CI to Github Actions

### PR DESCRIPTION
**Pull request by eht16**
Use Github Actions instead of external Travis CI service which recently enforced resource limits and so increase the risk of broken CI because of too few "credits".

I'm not completely sure about resource limits for Github Actions. According to https://docs.github.com/en/billing/managing-billing-for-github-actions/about-billing-for-github-actions, it seems there are 2000 free minutes per month.
OTOH the first sentence on this page says "GitHub Actions usage is free for both public repositories and self-hosted runners." while the table below says "2,000 minutes" for "GitHub Free".
According to https://github.community/t/for-public-repositories-is-there-a-monthly-limit-on-minutes/129017, there is no limit for public repositories.

Even if there would be a limit of 2000 minutes, this might fit for our use (for all repositories of the "geany" organization). On Travis CI, we have to manually ask their support to donate additional credits to our account each time the credits are exhausted.

From September 1st to October 17th, we used about 17500 credits (about 1750 build minutes) on Travis CI and we've got 8090 credits left to use. 

See also https://github.com/geany/www.geany.org/pull/33.
I would continue with migrating G-P once we agree on this switch and this one has been merged.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,161 @@
+#
+# Copyright:	2021, The Geany contributors
+# License:		GNU GPL v2 or later
+
+name: CI Build
+
+on: push
```
Comment (eht16) [inline .github/workflows/build.yml:7]: This is only for testing and I'll remove it before merge.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+      - name: Configure ccache
+        uses: actions/cache@v2
+        with:
+          path: ${{ env.CCACHE_DIR }}
+          key: ${{ runner.os }}-${{ github.job }}-ccache-${{ steps.ccache_cache_timestamp.outputs.timestamp }}
+          restore-keys: ${{ runner.os }}-${{ github.job }}-ccache-
+
+      - name: Show environment
+        run: env | sort
+        if: ${{ env.DEBUG == '1' }}
```
Comment (eht16) [inline .github/workflows/build.yml:65]: I'm not yet decided whether we should keep the "show environment" and "ccache statistics" steps. They helped while writing the workflow definition but might be just noisy in every day use.
For now, I made them disabled by default but activatable via the `DEBUG` variable.

**Comment by kugel-** [comment]
I agree on the switch. I would like to have CI for my meson branch again ;-)

If I see this correctly only Linux CI builds are covered for now. What are the plans for other platforms?

**Comment by kugel-** [comment]
Sorry, I have overlooked the mingw-part.

**Comment by eht16** [comment]
Merged to get progress on the migration to GH Actions. I'll continue with G-P soon.
Existing PRs probably will continue to use Travis CI until merged but this should be OK.

**Pull request merged**